pkg/controller/clusterautoscaler/clusterautoscaler_test.go

```diff
@@ -125,3 +132,120 @@ func TestAutoscalerArgs(t *testing.T) {
 		}
 	}
 }
+
+type MockReconciler struct {
+	getAutoscalerOk bool
+	gASerrType      error
+	configVersion   string
+	isDCCResult     bool
+	calledIsDCC     bool
+}
+
+func (r *MockReconciler) GetAutoscaler(_ *autoscalingv1alpha1.ClusterAutoscaler) (*appsv1.Deployment, error) {
+	dep := &appsv1.Deployment{}
+
+	if !r.getAutoscalerOk {
+		return nil, r.gASerrType
+	}
+	var m map[string]string
+	m = make(map[string]string)
+	dep.ObjectMeta.Annotations = m
```

Comment:
Why can't this just be `dep.ObjectMeta.Annotations = map[string]string{}`?

And/or:

```go
if dep.ObjectMeta.Annotations == nil {
    dep.ObjectMeta.Annotations = map[string]string{}
}
```

Reply:
I can collapse, but maps must be initialized.  Also, this should only be called once for each test, so if statement should be unnecessary.